packages/logger/src/Logger.ts

```diff
@@ -286,30 +284,43 @@ class Logger extends Utility implements ClassThatLogs {
 
       descriptor.value = (event, context, callback) => {
 
-        let initialPersistentAttributes: LogAttributes = {};
-        if (isClearStateEnabled) {
-          initialPersistentAttributes = { ...this.getPersistentLogAttributes() };
-        }
-
-        this.addContext(context);
-        let shouldLogEvent = undefined;
-        if ( options && options.hasOwnProperty('logEvent') ) {
-          shouldLogEvent = options.logEvent;
-        }
-        this.logEventIfEnabled(event, shouldLogEvent);
+        const initialPersistentAttributes = Logger.injectLambdaContextBefore(this, event, context, options);
```

Comment:
(optional) I find it hard to understand why `injectLambdaContextBefore()` return its initial state. It seems to do two things at once. 

Would it be easier to understand if we move the return logic out of the function? Something like this.

```suggestion
let initialPersistentAttributes = {};
if (options && options.clearState === true) {
  initialPersistentAttributes = logger.getPersistentLogAttributes();
}

Logger.injectLambdaContextBefore(/*...*/); // This method return nothing. 
```